### PR TITLE
Implement ECDSA Key Generation

### PR DESCRIPTION
**Pull request by Sean-Der**
Add generatedCertificateBits to KvsRtcConfiguration if you want the old behavior still

### CODE REVIEW

File: src/source/Dtls/Dtls.c [hunk 1]
```diff
@@ -77,26 +77,38 @@ STATUS dtlsTransmissionTimerCallback(UINT32 timerID, UINT64 currentTime, UINT64
...
-    CHK((pRsa = RSA_new()) != NULL, STATUS_CERTIFICATE_GENERATION_FAILED);
-    CHK(RSA_generate_key_ex(pRsa, certificateBits, pBne, NULL) != 0, STATUS_CERTIFICATE_GENERATION_FAILED);
-    CHK((EVP_PKEY_assign_RSA(*ppPkey, pRsa)) != 0, STATUS_CERTIFICATE_GENERATION_FAILED);
-    pRsa = NULL;
+        CHK((pRsa = RSA_new()) != NULL, STATUS_CERTIFICATE_GENERATION_FAILED);
+        CHK(RSA_generate_key_ex(pRsa, certificateBits, pBne, NULL) != 0, STATUS_CERTIFICATE_GENERATION_FAILED);
+        CHK((EVP_PKEY_assign_RSA(*ppPkey, pRsa)) != 0, STATUS_CERTIFICATE_GENERATION_FAILED);
+        pRsa = NULL;
+    } else {
+        CHK((eccGroup = OBJ_txt2nid("secp521r1")) != NID_undef, STATUS_CERTIFICATE_GENERATION_FAILED);
```
Comment (MushMal) [inline src/source/Dtls/Dtls.c]: So, which curve are we using then?

File: src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h [hunk 2]
```diff
@@ -659,6 +659,10 @@ typedef struct {
     // If unset GENERATED_CERTIFICATE_BITS will be used
     INT32 generatedCertificateBits;
 
+    // generateRSACertificate controls if an ECDSA or RSA certificate is generated. By default we generate an ECDSA
+    // certificate but some platforms may not support them.
+    BOOL generateRSACertificate;
```
Comment (MushMal) [inline src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h]: Not sure if there is a standard that talks about which curve and what bit-ness to use

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=h1) Report
> Merging [#230](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=desc) into [master](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/commit/5877c05a09f1e31daa05097c3e74750d0645b9d5&el=desc) will **decrease** coverage by `0.07%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #230      +/-   ##
==========================================
- Coverage   86.96%   86.89%   -0.08%     
==========================================
  Files          33       33              
  Lines        7820     7828       +8     
==========================================
+ Hits         6801     6802       +1     
- Misses       1019     1026       +7     
```


| [Impacted Files](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/source/Dtls/Dtls.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9EdGxzL0R0bHMuYw==) | `91.28% <0.00%> (-2.26%)` | :arrow_down: |
| [src/source/PeerConnection/PeerConnection.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9QZWVyQ29ubmVjdGlvbi9QZWVyQ29ubmVjdGlvbi5j) | `93.94% <0.00%> (ø)` | :arrow_up: |
| [src/source/Ice/IceAgent.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9JY2UvSWNlQWdlbnQuYw==) | `93.88% <0.00%> (-0.11%)` | :arrow_down: |
| [src/source/PeerConnection/SessionDescription.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9QZWVyQ29ubmVjdGlvbi9TZXNzaW9uRGVzY3JpcHRpb24uYw==) | `83.82% <0.00%> (ø)` | :arrow_up: |
| [src/source/Ice/IceAgentStateMachine.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9JY2UvSWNlQWdlbnRTdGF0ZU1hY2hpbmUuYw==) | `83.33% <0.00%> (+0.29%)` | :arrow_up: |
| [src/source/PeerConnection/JitterBuffer.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9QZWVyQ29ubmVjdGlvbi9KaXR0ZXJCdWZmZXIuYw==) | `95.48% <0.00%> (ø)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=footer). Last update [a494c5f...d719473](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: src/source/Dtls/Dtls.c (hunk 1 above)
Comment (Sean-Der) [reply src/source/Dtls/Dtls.c]: Fixed, `NIST P-256`

File: src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h (hunk 2 above)
Comment (Sean-Der) [reply src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h]:
```
All Implementations MUST support DTLS 1.2 with the
   TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256 cipher suite and the P-256
   curve [FIPS186].  Earlier drafts of this specification required DTLS
   1.0 with the cipher suite TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA, and
   at the time of this writing some implementations do not support DTLS
   1.2; endpoints which support only DTLS 1.2 might encounter
   interoperability issues.  The DTLS-SRTP protection profile
   SRTP_AES128_CM_HMAC_SHA1_80 MUST be supported for SRTP.
   Implementations MUST favor cipher suites which support (Perfect
   Forward Secrecy) PFS over non-PFS cipher suites and SHOULD favor AEAD
   over non-AEAD cipher suites.
```

**Comment by Sean-Der** [comment]
I tested with Safari, FireFox, Chrome and Android. Haven't been able to get an iOS device yet.

File: src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h (hunk 2 above)
Comment (MushMal) [reply src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h]: Cool. MINOR: Could you include the link in the comment?

**Comment by alexw91** [comment]
# [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=h1) Report
> Merging [#230](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=desc) into [master](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/commit/5877c05a09f1e31daa05097c3e74750d0645b9d5&el=desc) will **increase** coverage by `0.01%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=tree)

```diff
@@            Coverage Diff             @@
##           master     #230      +/-   ##
==========================================
+ Coverage   86.96%   86.98%   +0.01%     
==========================================
  Files          33       33              
  Lines        7820     7828       +8     
==========================================
+ Hits         6801     6809       +8     
  Misses       1019     1019              
```


| [Impacted Files](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=tree) | Coverage Δ | |
|---|---|---|
| [src/source/PeerConnection/PeerConnection.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9QZWVyQ29ubmVjdGlvbi9QZWVyQ29ubmVjdGlvbi5j) | `93.94% <0.00%> (ø)` | :arrow_up: |
| [src/source/Ice/IceAgent.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9JY2UvSWNlQWdlbnQuYw==) | `93.88% <0.00%> (-0.11%)` | :arrow_down: |
| [src/source/PeerConnection/SessionDescription.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9QZWVyQ29ubmVjdGlvbi9TZXNzaW9uRGVzY3JpcHRpb24uYw==) | `83.82% <0.00%> (ø)` | :arrow_up: |
| [src/source/Ice/IceAgentStateMachine.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9JY2UvSWNlQWdlbnRTdGF0ZU1hY2hpbmUuYw==) | `83.33% <0.00%> (+0.29%)` | :arrow_up: |
| [src/source/PeerConnection/JitterBuffer.c](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230/diff?src=pr&el=tree#diff-c3JjL3NvdXJjZS9QZWVyQ29ubmVjdGlvbi9KaXR0ZXJCdWZmZXIuYw==) | `95.48% <0.00%> (ø)` | :arrow_up: |

------

[Continue to review full report at Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=footer). Last update [a494c5f...766c002](https://codecov.io/gh/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/pull/230?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Pull request merged**